misk/src/main/kotlin/misk/web/actions/UpstreamResourceInterceptor.kt

```diff
@@ -0,0 +1,69 @@
+package misk.web.actions
+
+import misk.web.NetworkChain
+import misk.web.NetworkInterceptor
+import misk.web.Response
+import misk.web.toMisk
+import misk.web.toOkHttp3
+import okhttp3.HttpUrl
+
+class UpstreamResourceInterceptor(
+  private val mappings: MutableList<out Mapping>
+) : NetworkInterceptor {
+  @Suppress("UNUSED_PARAMETER")
+  override fun intercept(chain: NetworkChain): Response<*> {
+    val requestSegments = chain.request.url.pathSegments()
+    var matchedMapping: Mapping? = null
+
+    for (mapping in this.mappings) {
+      if (!pathSegmentsMatch(mapping.localPathPrefix.drop(1).dropLast(1).split('/'), requestSegments)) continue
```

Comment:
I suspect the drops are a signal that something is wrong here. Can you just do

chain.request.url.encodedPath().startsWith(mapping.localPathPrefix)

Reply:
ie. match on the strings rather than the list of path segments

Reply:
Yeah, that makes a lot more sense 👍